### ISSUE TITLE
Download firefox nightly Crash

### ISSUE THREAD

**Issue by PsykoRider** [Tobi823/ffupdater#99]
Stacktrace:
```
java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.String android.net.Uri.getLastPathSegment()' on a null object reference
	at android.app.DownloadManager.enqueue(DownloadManager.java:1626)
	at de.marmaro.krt.ffupdater.download.DownloadManagerUtil.enqueue(DownloadManagerUtil.kt:37)
	at de.marmaro.krt.ffupdater.InstallActivity$Companion.enqueuingDownload(InstallActivity.kt:319)
	at de.marmaro.krt.ffupdater.InstallActivity$State._init_$enqueuingDownload(InstallActivity.kt:202)
	at de.marmaro.krt.ffupdater.InstallActivity$State.access$_init_$enqueuingDownload(InstallActivity.kt:195)
	at de.marmaro.krt.ffupdater.InstallActivity$State$6.invoke(InstallActivity.kt:202)
	at de.marmaro.krt.ffupdater.InstallActivity$State$6.invoke(InstallActivity.kt:202)
	at de.marmaro.krt.ffupdater.InstallActivity$restartStateMachine$1.invokeSuspend(InstallActivity.kt:169)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
	at android.os.Handler.handleCallback(Handler.java:938)
	at android.os.Handler.dispatchMessage(Handler.java:99)
	at android.os.Looper.loop(Looper.java:236)
	at android.app.ActivityThread.main(ActivityThread.java:7876)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:656)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:967)
```
Device information:
| key | value |
| --- | --- |
| version name | 74.5.3 |
| version code | 90 |
| build type | release |
| SDK version | 30 |
| android release | 11 |
| board | lmi |
| brand | POCO |
| device | lmi |
| manufacturer | Xiaomi |
| model | POCO F2 Pro |
| product | lmi_eea |
| SOC manufacturer | / |
| SOC model | / |
| supported ABIs | arm64-v8a, armeabi-v7a, armeabi |
| tags | release-keys |
| build time | 1637375224000 |

**Comment by Tobi823** [comment]
When does the error occur? When downloading a specific app?

**Comment by Tobi823** [comment]
Can you try this new version?
[ffupdater-release.zip](https://github.com/Tobi823/ffupdater/files/7794621/ffupdater-release.zip)
(You have to decompress this ZIP-archiv and uninstall the FFUpdater version from F-Droid)

**Comment by Tobi823** [comment]
For developers:

I was puzzled at first why this error occurs. The Uri-object I pass to the enqueue-method is definitely not null.

But then I've looked into the source code of the DownloadManager::enqueue-method (https://cs.android.com/android/platform/superproject/+/master:frameworks/base/core/java/android/app/DownloadManager.java;l=1117):

```
    private final ContentResolver mResolver;
    ....

    public DownloadManager(Context context) {
        mResolver = context.getContentResolver();
        ....
    }

    public long enqueue(Request request) {
        ContentValues values = request.toContentValues(mPackageName);
        Uri downloadUri = mResolver.insert(Downloads.Impl.CONTENT_URI, values);
        long id = Long.parseLong(downloadUri.getLastPathSegment());
        return id;
    }
```

I suspect that mResolver returns a null object. Why? I don't know - maybe I'm calling the enqueue-method from a wrong context.

And that's my current attempt: I never store the instance of DownloadManager as a variable. I always request the current instance with `context.getSystemService()`

**Comment by Tobi823** [comment]
Should be fixed in version 75.0.0

**Comment by PsykoRider** [comment]
I'm sorry. Just I see now the email. Thank you for the fix solution.
